### ISSUE TITLE
DetailsList: make column resizability keyboard accessible

### ISSUE THREAD

**Issue by Lovakumar** [microsoft/fluentui#4947]
### Bug Report

- __Package version(s)__: (fill this out)
- __Browser and OS versions__: 
OS: Windows 10 RS3
OS version: build 16299.371
Browser: Microsoft edge browser, IE11 browser
Edge version: Microsoft Edge 41.16299.371.0
Firefox version: 60.0(32-bit)
VSTS Version: 20180510T050959/_cssbundles/Default
                       20180512T183414

#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? (No)

Requested priority: (Normal) 

Products/sites affected: (if applicable)

#### Describe the issue:

Splitter is not accessible by keyboard in DetailsList control.

#### Actual behavior:

Splitter is not accessible by keyboard in DetailsList control. We can't resize columns using key board.

#### Expected behavior:

 Using keyboard we should be to resize columns

**Comment by peterdur-msft** [comment]
This also came up in the Wipro accessibility test pass on Keros for Web. We look forward to picking up the fix.

**Comment by ThomasMichon** [comment]
@lovakumar @peterdur-msft @betrue-final-final How should we make the splitter keyboard-navigable? I had previously made it so, but it was confusing when using the arrow keys to navigate between column headers that focus would get 'hung up' on the splitter. *If* we provide a way to navigate to it, the splitter already has handling for keyboard eventing to move it.

**Comment by peterdur-msft** [comment]
Checked in with teammate and got the following suggestions:

Since this is a grid, it should always support keyboard interaction pattern for a grid, i.e. using arrow keys should allow navigating to the next/previous columns.

There is no specific accessible interaction pattern for this scenario per se but we can try adding keyboard support for moving the splitter using different key combinations:

Option-1: Pressing “enter” key to enable the splitter position movement and use the arrow keys to move its position (Similar to the keyboard interaction in  https://www.w3.org/TR/wai-aria-practices-1.1/examples/grid/dataGrids.html#ex2_label )

Option-2: Using (ctrl + arrow) keys to move the splitter once it receives focus

In both the cases, we should let the user know that additional keyboard interaction option is available

**Comment by dzearing** [comment]
@peterdur-msft @ThomasMichon mentioned above that it was an odd experience to make the "splitter" area focusable. That is, starting with col1, pressing right arrow would move focus like so:

col1 > splitter > col2 > splitter > col3 > splitter

...which ends up confusing users, especially when some columns can be resized and others can't.

So @betrue-final-final what is the design here?

My preference would be to not change the design of keeping the arrow targets only going between columns... but we *could* make something like ctrl/command arrowing resize the column by X pixels.

It could be a way to make it resizable via keyboard, though maybe a little undiscoverable.

**Comment by betrue-final-final** [comment]
Wow, it's been awhile since I looked at this. @dzearing, I get where you're coming from about the extra tab stop. We can do Ctl/Cmd + arrow to resize and try that. That should be ok for the short term.

**Comment by cschlechty** [comment]
I think it is agreed that we should do this work, we just need to lock down what the shortcut will be and get somebody assigned.

**Comment by smhigley** [comment]
This came up in a separate bug -- I deifnitely agree with @dzearing that default arrow nav should not go to resize handles.

One suggestion that came up was to have a toggle button (could be only visible on focus, like a skip link, or part of a context menu) that turns on resize handles and places them in the tab order.

**Comment by cschlechty** [comment]
Does this need to be a mode? When focuses on a header, can we just provided key combination that expands left and right? That seems like the easiest.

**Comment by smhigley** [comment]
That would work! I was thinking with a button it might be more discoverable, but I don't actually know which would work better in practice :)

**Comment by vanessgo** [comment]
What is your eta on fixing this issue?
